core/src/main/java/hudson/Util.java

```diff
@@ -1755,4 +1785,15 @@ public static void closeAndLogFailures(@CheckForNull Closeable toClose, @Nonnull
      */
     @Restricted(value = NoExternalUse.class)
     static boolean GC_AFTER_FAILED_DELETE = SystemProperties.getBoolean(Util.class.getName() + ".performGCOnFailedDelete");
+
+    /**
+     * If this flag is true, native implementations of {@link FilesPath#chmod}
+     * and {#IOUtils#mode} are used instead of NIO.
+     * <p>
+     * This should only be enabled if the setgid/setuid/sticky bits are
+     * intentionally set on the Jenkins installation and they are being
+     * overwritten by Jenkins erroneously.
+     */
+    @Restricted(value = NoExternalUse.class)
+    public static boolean NATIVE_CHMOD_MODE = SystemProperties.getBoolean(Util.class.getName() + ".useNativeChmodAndMode");
```

Comment:
I hate to introduce another system property... Is there a better way to this?

Reply:
Fine for now. I'd guess at some point we will create GUI configurer for System properties. It is my TODO list for years :(

Reply:
FindBugs should als grumble that the field is not final. Does not matter for now, FindBugs cleanup still needs to be done in the core